assign-doi-datacite/src/test/java/no/unit/nva/doi/datacite/customerconfigs/CustomerConfigsExtractorImplTest.java

```diff
@@ -0,0 +1,73 @@
+package no.unit.nva.doi.datacite.customerconfigs;
+
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static no.unit.nva.testutils.RandomDataGenerator.randomUri;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.nio.file.Path;
+import no.unit.nva.stubs.FakeSecretsManagerClient;
+import nva.commons.core.ioutils.IoUtils;
+import nva.commons.core.paths.UriWrapper;
+import nva.commons.secrets.SecretsReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CustomerConfigsExtractorImplTest {
+
+    private static final String SECRET_NAME = "someSecretName";
+    private static final String SECRET_KEY = "someSecretKey";
+    private SecretsReader secretsReader;
+
+    private CustomerConfigExtractorImpl customerConfigExtractor;
+
+    @BeforeEach
+    void setup() {
+        var fakeSecretsManagerClient = new FakeSecretsManagerClient();
+        fakeSecretsManagerClient.putSecret(SECRET_NAME, SECRET_KEY, getValidSecretString());
+        this.secretsReader = new SecretsReader(fakeSecretsManagerClient);
+        this.customerConfigExtractor = new CustomerConfigExtractorImpl(secretsReader,
+                                                                       SECRET_NAME,
+                                                                       SECRET_KEY);
+    }
+
+    @Test
+    void shouldThrowExceptionIfSecretReaderDosNotContainSecret() {
```

Comment:
typo in DosNotContain...

Also, what makes this have a missing secret. It seems valid to me?

Reply:
Renamed to "shouldThrowExceptionWhenRetrievingCustomerConfigIfSecretReaderDoesNotContainCustomer"